res/shaders/imgui_ps_4_0_hdr.hlsl

```diff
@@ -0,0 +1,86 @@
+Texture2D t0 : register(t0);
+SamplerState s0 : register(s0);
+
+cbuffer PSPushConstants : register(b0)
+{
+	uint  back_buffer_format               : packoffset(c4.x);
+	uint  back_buffer_color_space          : packoffset(c4.y);
+	float hdr_overlay_brightness           : packoffset(c4.z);
+	int   overwrite_overlay_color_space_to : packoffset(c4.w);
+};
+
+#define RGB10A2_UNORM 24
+#define BGR10A2_UNORM 1111970609
+#define RGBA16_FLOAT  10
+
+#define CSP_SCRGB 2
+#define CSP_HDR10 3
+
+#define OVERWRITE_CSP_TO_HDR10 1
+#define OVERWRITE_CSP_TO_SCRGB 2
+#define OVERWRITE_CSP_TO_SDR   3
+
+static const float PQ_m1 =  0.1593017578125;
+static const float PQ_m2 = 78.84375;
+static const float PQ_c1 =  0.8359375;
+static const float PQ_c2 = 18.8515625;
+static const float PQ_c3 = 18.6875;
+
+static const float3x3 bt709_to_bt2020 = {
+	0.627225305694944,  0.329476882715808,  0.0432978115892484,
+	0.0690418812810714, 0.919605681354755,  0.0113524373641739,
+	0.0163911702607078, 0.0880887513437058, 0.895520078395586};
+
+float3 pq_inverse_eotf(float3 col)
+{
+	float3 col_pow_m1 = pow(col, PQ_m1);
+
+	return pow((PQ_c1 + PQ_c2 * col_pow_m1) / (1.f + PQ_c3 * col_pow_m1), PQ_m2);
+}
+
+//HDR10 (PQ + BT.2020 primaries)
+float4 to_pq(float4 col)
+{
+	return float4(pq_inverse_eotf(mul(bt709_to_bt2020, pow(col.rgb, 2.2f)) * hdr_overlay_brightness / 10000.f), col.a);
+}
+
+//scRGB (linear + BT.709/sRGB primaries)
+float4 to_scrgb(float4 col)
+{
+	return float4(pow(col.rgb, 2.2f) * hdr_overlay_brightness / 80.f, col.a);
+}
+
+void main(float4 vpos : SV_POSITION, float4 vcol : COLOR0, float2 uv : TEXCOORD0, out float4 col : SV_TARGET)
+{
+	col = t0.Sample(s0, uv);
```

Comment:
I'm curious how textures interact with this. They will be in sRGB color space coming out of here and only the color space of the overlay colors mixed into that below are changed. Does that hold up? E.g. what happens when looking at the Statistics tab in HDR.